packages/pigeon/lib/dart_generator.dart

```diff
@@ -442,10 +442,15 @@ $resultAt != null
                   } else {
                     indent.writeln('final $returnType output = $call;');
                   }
+
                   const String returnExpression = 'output';
+                  final String nullability =
+                      func.returnType.isNullable ? '?' : '';
+                  final String enumOutput =
+                      isEnum(root, func.returnType) ? '$nullability.index' : '';
```

Comment:
It's weird to call this `enumOutput` when it's usually not a enum. Maybe `valueExtraction`?

Reply:
I agree, but it is an exclusively enum string, and the name makes it clear that it is. enumExtraction might be better though. I'm open to whatever makes sense though.

Reply:
I missed that it's empty in the non-enum case; I was thinking it was wrapping and replacing, rather than additive. Your naming is fine given that, sorry for the noise.